### ISSUE TITLE
Load deadlock in latest master

### ISSUE THREAD

**Issue by asvetliakov** [asvetliakov/vscode-neovim#340]
Alright, so I played around more trying to corner the issue.
I'm just going to focus on the issue of neovim not loading properly and leave the cursor jump issue for later.

Unfortunately, it seems like there isn't a single cause for neovim to not load properly. With a 300 line fully-commented out init.vim, and all my vscode extensions enabled, it locks up. With all the vscode extensions disabled, it works. However, by slowly re-enabling extensions, the issue has come back with multiple different plugins. What I mean is that I have about 3 plugins that when they are all disabled, it works fine, but when I enable any one of them (gitlens, wakatime, code-autocomplete), the issue comes back. In fact, when I just enable wakatime, and restart vscode many times, it actually only fails to load neovim *half* of the time. I have also encountered the issue with all those 3 plugins disabled, and also sometimes have not had the issue with all plugins enabled.

So it seems like there is some weird timing/race condition going on. It seems like the longer vscode takes to load, the more likely the lockup is to happen, and the longer init.vim takes to parse, the less likely the lockup is to happen. I'm not sure exactly how to help next.

Some information/observations:
 - Pop!_OS 20.04 LTS x86_64
 - vscode 1.49.0-insiders
 - NVIM v0.5.0-dev
 - vscode-neovim compiled from source from master (as of writing this)
 - everything works fine in console neovim
 - works fine with plugins and init.vim enabled, with vscode-neovim 0.0.52
 - the issue still happens with init.vim completely deleted (as opposed to commented out)

In 0.0.52, the cursor seems to go to block mode on boot (how I can tell neovim has loaded) much later than in 0.0.60. It seems to be ready only after everything has loaded, including semantic highlighting and lsp symbol tree. In 0.0.60, the cursor goes to block mode and is responsive to vim input before everything has finished loading.

Hopefully this gives enough information to pin down the issue, I'm not sure :smile:.
Do you want me to create a new issue for this? It directly relates to refactor but you have already merged and closed this PR :shrug:

_Originally posted by @theol0403 in https://github.com/asvetliakov/vscode-neovim/pull/334#issuecomment-687690967_

**Comment by asvetliakov** [comment]
@theol0403 thank you for writing up. i'll try to look

**Comment by asvetliakov** [comment]
@theol0403

Feel free to reopen if the problem still happens

**Comment by theol0403** [comment]
Thanks, that solved it!

**Comment by sisrfeng** [comment]
cursor jump problem appers when I enable gitlens
